### PR TITLE
Daniel Qu Firmware 103 Homework

### PR DESCRIPTION
**Pull request by danielq987**

### CODE REVIEW

File: projects/button_adc/src/main.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#include <stdbool.h>
```
Comment (AveryChiu64) [inline projects/button_adc/src/main.c:1]: Do you need <stdbool.h>?

File: projects/button_adc/src/main.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#include "wait.h"
+
+// Interrupt callback
+static void prv_adc_callback(GpioAddress *address, void *context) {
+  GpioAddress *adc_address = context;
+
+  uint16_t adc_reading = 0;
+  adc_read_raw_pin(*adc_address, &adc_reading);
+
+  LOG_DEBUG("Read ADC value");
```
Comment (AveryChiu64) [inline projects/button_adc/src/main.c:16]: We should show the actual ADC value that we read

File: projects/button_adc/src/main.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    .port = GPIO_PORT_A,  //
+    .pin = 6,             //
+  };
+
+  GpioAddress button_address = {
+    .port = GPIO_PORT_B,  //
+    .pin = 2,             //
+  };
+
+  GpioSettings gpio_settings = {
```
Comment (AveryChiu64) [inline projects/button_adc/src/main.c:31]: You need separate settings for the ADC pin and the button pin. They should be different. One reads an analog signal. Another reads a digital one.

File: projects/button_adc/src/main.c [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  adc_init(ADC_MODE_SINGLE);
+
+  // Pin Inits
+  gpio_init_pin(&adc_address, &gpio_settings);
+  gpio_init_pin(&button_address, &gpio_settings);
+
+  adc_set_channel_pin(adc_address, true);
+
+  // Interrupt Code
+  gpio_it_register_interrupt(&button_address, &interrupt_settings, INTERRUPT_EDGE_RISING_FALLING,
```
Comment (AveryChiu64) [inline projects/button_adc/src/main.c:55]: Why is it edge rising and falling?

**Comment by mitchellostler** [comment]
Hey Daniel, I thought this PR was still waiting on changes my bad. Looks good, you're good to move on

**Pull request closed without merging**